src/ui/osx/TogglDesktop/Feature/Permission/SystemPermissionManager.swift

```diff
@@ -0,0 +1,169 @@
+        case screenRecording
+
+        var urlScheme: String {
+            switch self {
+            case .screenRecording:
+                return "x-apple.systempreferences:com.apple.preference.security?Privacy_ScreenCapture"
+            }
+        }
+    }
+
+    private struct Constants {
+        static let ScreenRecordingPermissionGranted = "ScreenRecordingPermissionGranted"
+    }
+
+    static let shared = SystemPermissionManager()
+
+    // MARK: Public
+
+    func isGranted(_ permission: Permission) -> Bool {
+        switch permission {
+        case .screenRecording:
+            if #available(OSX 10.15, *) {
+                if isAlreadyRequestSystemPermission(permission) {
+                    return canRecordScreen()
+                }
+                return false
+            }
+            return true // macOS < 10.15 are granted by default
+        }
+    }
+
+    func grant(_ permission: Permission) {
+        // Skip if it's already granted
+        guard !isGranted(permission) else { return }
+
+        switch permission {
+        case .screenRecording:
+            if #available(OSX 10.15, *) {
+
+                // Show alert to instruct the user to manually grant the permission
+                if isAlreadyRequestSystemPermission(permission) {
+                    presentScreenRecordingAlert {
+                        tryOpeningSystemPreference(for: permission)
+                    }
+                } else {
+                    // Trigger the system alert once
+                    triggerScreenRecordingPermissionAlert()
+                }
+            }
+        }
+    }
+}
+
+// MARK: Private
+
+extension SystemPermissionManager {
+
+    private func presentScreenRecordingAlert(_ complete: () -> Void) {
+        let alert = NSAlert()
+        alert.messageText = "Screen Recording permission not granted!"
+        alert.informativeText = "To get the Windows Name properly for the Timeline, TogglDesktop needs to be granted the Screen Recording permission in Security & Privacy in System Preferences .\n\nPlease open System Preferences -> Security & Privacy -> Privacy Tab -> Select Screen Recording and enable TogglDesktop app."
+        alert.alertStyle = .warning
+        alert.addButton(withTitle: "Open")
+        alert.addButton(withTitle: "Later")
+
+        let result = alert.runModal()
+        if result == .alertFirstButtonReturn {
+            complete()
+        }
+    }
+
+    private func isAlreadyRequestSystemPermission(_ permission: Permission) -> Bool {
+        switch permission {
+        case .screenRecording:
+            return UserDefaults.standard.bool(forKey: Constants.ScreenRecordingPermissionGranted)
+        }
+    }
+
+    private func tryOpeningSystemPreference(for permission: Permission) {
+
+        // Open the Screen Recording tab
+        // https://stackoverflow.com/a/48139877/3127477
+        if let url = URL(string: permission.urlScheme) {
+            NSWorkspace.shared.open(url)
```

Comment:
Some hack to achieve:
- Open the System Preference -> Security -> Privacy Tab and navigate to ScreenRecording section.